presto-hive/src/main/java/com/facebook/presto/hive/HivePageSink.java

```diff
@@ -212,12 +216,24 @@ public void abort()
 
     private void doAbort()
     {
+        Map<HiveWriter, Exception> exceptions = new HashMap<>();
         for (HiveWriter writer : writers) {
             // writers can contain nulls if an exception is thrown when doAppend expends the writer list
             if (writer != null) {
-                writer.rollback();
+                try {
+                    writer.rollback();
+                }
+                catch (Exception e) {
+                    exceptions.put(writer, e);
+                }
             }
         }
+        if (!exceptions.isEmpty()) {
+            exceptions.entrySet().forEach(
+                    entry -> log.warn("exception '%s' while rollback on %s", entry.getValue(), entry.getKey())
+            );
+            throw new PrestoException(HIVE_WRITER_CLOSE_ERROR, "Error rolling back write to Hive");
```

Comment:
Put one of the exceptions as the cause for this. It would be nice to have the file name for that one as well.

@dain, @electrum, and I considered putting all of them in suppressed, and decided not to that. But we feel like that is not likely going to give more information. And when we add a list to suppressed, we have to be careful about not having too many of them, which is additional hassle.